### DISCUSSION TITLE
On an brand new deployment I receive an error: "Received 404 on websocket"

### DISCUSSION THREAD

**Discussion by jvieten** [strimzi/strimzi-kafka-operator#9666] [General]
I have deployed the Strimzi operator 0.39.0 to a kubernetes cluster.  In the log of the deployment I see:   

  `Main:209 - Cluster Operator verticle in namespace kafka-connect failed to start
io.fabric8.kubernetes.client.KubernetesClientException: Received 404 on websocket. Failure executing: GET at: https://10.10.32.1:443/apis/kafka.strimzi.io/v1beta2/namespaces/kafka-connect/kafkanodepools?allowWatchBookmarks=true&watch=true. Message: Not Found`.

The Address https://10.10.32.1:443 is the service of kubernetes cluster.
Would somebody  be able to give me a hint on what might be going wrong?

**Comment by scholzj** [comment]
That suggests that the CRD is not installed - in this case the KafkaNodePool CRD. So you should make sure it is installed.

**Reply by jvieten** [reply]
thanks alot that was it! 

Background I am installing strimzi by terraform.
`
resource "helm_release" "strimzi"  {

  chart = "strimzi-kafka-operator"
  name = "strimzi"
  version = "0.39.0"
  repository = "oci://quay.io/strimzi-helm"
  namespace  = "strimzi"
  wait       = true 
}`

To get rid of the error I had to completely uninstall  the resource by deleting the terraform "helm_release" resource,  then "terraform apply" and then adding it back and then "terraform apply" again. 

This failure happens **at first** on all my test, stage and prod environments , so I am guessing that something is not working correctly on a **first time install** with terraform and its helm or kubernetes  providers.  

Maybe others had this problem also and know what the real reason is?

But I am so glad that it works now!

**Comment by AlexsandroRotundo** [comment]
The problem is related to a new feature introduced by strimzi in the version 0.39. As stated by the official documentation the KafkaNodePools 
feature gate is enabled by default. I simply solved the issue disabling the feature in the values.yaml file. 
The official documentation about the disable below:
Disabling the KafkaNodePools feature gate
To disable the KafkaNodePools feature gate, specify -KafkaNodePools in the STRIMZI_FEATURE_GATES environment variable in the Cluster Operator configuration. The Kafka custom resource using the node pools must also have the annotation strimzi.io/node-pools: enabled

The STRIMZI_FEATURE_GATES environment variable is featureGates.

The link to the official documentation: https://strimzi.io/docs/operators/latest/deploying.html#ref-operator-kafka-node-pools-feature-gate-str

**Reply by scholzj** [reply]
I guess that would just confirm what I suggested about the missing CRD.

**Reply by AlexsandroRotundo** [reply]
In my case the problem was due to a couple of things. The missing CRD KafkaNodePool and the KafkaNodePool feature gate enabled by default.

**Reply by scholzj** [reply]
But why would you need to disable the feature gate if you installed the CRD? The feature gate would open a watch for the KafkaNodePools, but not do anything else unless you intentionally start using NodePools.

**Reply by AlexsandroRotundo** [reply]
I've not installed the KafkaNodePool CRD. I've never used KafkaNodePool object in my cluster. I had the problem described and only disabling it my strimzi operator went up&running. Maybe, in the future I'll study the feature and I'll enable the KafkaNodePool in my operator configuration.
